tests/functional/test_install_cleanup.py

```diff
@@ -132,7 +132,7 @@ def test_cleanup_prevented_upon_build_dir_exception(script, data):
     result = script.pip(
         'install', '-f', data.find_links, '--no-index', 'simple',
         '--build', build,
-        expect_error=True,
+        expect_error=True, expect_temp=True,
```

Comment:
The cache should be cleaning up after itself even if this happens?

Reply:
@takluyver Can you explain this?

Reply:
Sorry, I can't remember the details. At some point some tests failed, and that seemed to be the right thing to make them work.

Reply:
Perhaps the PR does not clean up the cache as advertised?  What is the affect of this parameter?

Reply:
AFAICT, this parameter means that the script run is expected to leave some temporary files, which is not what we want.

Reply:
What we can do it turn the following line into a `with` statement and set the wheelcache as a context manager:

https://github.com/pypa/pip/blob/6e2391a92c82e5182a6127ef79e41f8b0beaf2ab/src/pip/_internal/commands/install.py#L222

Is there any other location where the wheelcache is initialized?

Reply:
@xoviat -- a simple grep tells me there are 6 places.

Reply:
@pradyunsg So this test is named `cleanup_prevented_on_build_dir_exception`, and if you look at the control flow on `install`, it makes sense why the directory would not cleaned up. I attempted to fix that with the contextmanager, but your proposed solution necessitates that temporary files will be created here.

Reply:
Cool. That means that this change in tests corresponds to a change in the expected behaviour. :)

PS: 41b83f8 was not the change I was suggesting. I thought there was no call to cleanup in finally.